parachain/runtime/kintsugi/src/dex.rs

```diff
@@ -0,0 +1,247 @@
+    type PoolCurrencySymbolLimit = StringLimit;
+    type PalletId = DexStablePalletId;
+    type WeightInfo = ();
+}
+
+impl dex_swap_router::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type StablePoolId = StablePoolId;
+    type Balance = Balance;
+    type CurrencyId = CurrencyId;
+    type NormalAmm = DexGeneral;
+    type StableAMM = DexStable;
+    type WeightInfo = ();
+}
+
+pub struct DexSetup;
+
+fn create_pair_and_update_rewards(asset_0: CurrencyId, asset_1: CurrencyId, rewards: Balance) {
+    // TODO: should we bootstrap?
+    if let Err(err) = DexGeneral::create_pair(
+        RuntimeOrigin::root(),
+        asset_0,
+        asset_1,
+        30, // 0.3%
+    ) {
+        log::error!("Could not create pair: {:?}", err);
+    } else {
+        if let Err(err) = Farming::update_reward_schedule(
+            RuntimeOrigin::root(),
+            CurrencyId::join_lp_token(asset_0, asset_1).expect("currencies are valid; qed"),
+            CurrencyId::Token(KINT),
+            132480, // 60 * 24 * 92 (approx three months)
+            rewards,
+        ) {
+            log::error!("Could not update rewards: {:?}", err);
+        }
+    }
+}
+
+impl OnRuntimeUpgrade for DexSetup {
+    fn on_runtime_upgrade() -> Weight {
+        create_pair_and_update_rewards(
+            CurrencyId::Token(KBTC),
+            CurrencyId::Token(KSM),
+            45000000000000000, // 45,000 KINT
+        );
+
+        create_pair_and_update_rewards(
+            CurrencyId::Token(KBTC),
+            CurrencyId::ForeignAsset(3), // USDT
+            40000000000000000,           // 40,000 KINT
+        );
+
+        create_pair_and_update_rewards(
+            CurrencyId::Token(KSM),
+            CurrencyId::Token(KINT),
+            35000000000000000, // 35,000 KINT
+        );
+
+        Default::default()
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        ensure!(
+            dex_general::PairStatuses::<Runtime>::iter().collect::<Vec<_>>().len() == 0,
+            "Should not have pools"
+        );
+        Ok(Vec::new())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
+        ensure!(
+            dex_general::PairStatuses::<Runtime>::iter().collect::<Vec<_>>().len() == 3,
+            "Should have pools"
+        );
+        Ok(())
+    }
+}
+
+pub struct LoansSetup;
+
+fn add_and_activate_market(token: CurrencyId, market: loans::Market<Balance>) {
+    if let Err(err) = Loans::add_market(RuntimeOrigin::root(), token, market) {
+        log::error!("Could not add market: {:?}", err);
+    } else {
+        // TODO: add rewards?
+        if let Err(err) = Loans::activate_market(RuntimeOrigin::root(), token) {
+            log::error!("Could not activate market: {:?}", err);
+        }
+    }
+}
+
+impl OnRuntimeUpgrade for LoansSetup {
+    fn on_runtime_upgrade() -> Weight {
```

Comment:
@daniel-savu please review lending market construction. @philippb90 it was also not clear to be what the `collateral_threshold` and `liquidation_threshold` should be set to for each instantiation.